Pldeschaetzen/bac 761 updating mcp server

hhertoghe opened the pull request.

pldesch-chift committed:
chore: update dependencies and refactor consumer tool registration

- Bump `chift` version to 0.3.11 in `pyproject.toml` and `uv.lock`.
- Refactor `register_tools` to `register_consumer_tools` for clarity.
- Enhance `import_toolkit_functions` to support automatic injection of `consumer_id`.
- Update type hints and return types in various functions for better clarity and type safety.
- Add new utility function to retrieve connection types based on consumer ID.

pldesch-chift committed:
refactor(mcp): restructure main application logic and remove unused components

- Transition to Fastmcp instead of mcp
- Refactor `__main__.py` to utilize asynchronous functions for improved performance.
- Replace `FastMCP` import path and update related function signatures.
- Remove unused `toolkit.py`, `importer.py`, and `openapi.json` files to clean up the codebase.
- Consolidate connection type retrieval into the `utils` module.
- Update configuration handling and improve logging for better traceability.

pldesch-chift committed:
refactor(mcp): enhance configuration and middleware integration

- Replace direct configuration access with a structured `Chift` class for better encapsulation.
- Update `configure_chift` function to accept a `Chift` instance, improving type safety.
- Integrate middleware components into the MCP initialization for enhanced functionality.
- Remove unused OpenAPI parsing utilities and the `openapi.json` file to streamline the codebase.
- Refactor tool customization to support hiding consumer IDs and pagination more effectively.

pldesch-chift committed:
test: add initial test for get_mcp function

- Introduced a new test file for the MCP module.
- Added an asynchronous test for the `get_mcp` function to verify tool retrieval.

pldesch-chift committed:
feat: enhance project configuration and testing setup

- Added `fastmcp` as a dependency for improved functionality.
- Introduced `pytest-asyncio` for asynchronous testing support.
- Configured pytest options for better test discovery and organization.
- Removed unused `http.py` file and refactored utility functions for clarity.
- Added mock fixtures for `FastMCP` and `Tool` to facilitate testing.
- Removed outdated tests related to the importer and utilities, streamlining the test suite.

pldesch-chift committed:
refactor(mcp): enhance main application structure and remove obsolete test file

- Refactored `__main__.py` to integrate new OpenAPI components and improve MCP initialization.
- Introduced `mcp_component_fn` for better handling of OpenAPI resources.
- Removed obsolete `__test__.py` file to streamline the codebase.
- Updated `http_client.py` and `middleware.py` for improved code clarity and consistency.
- Enhanced utility functions in `utils.py` for better configuration management.

pldesch-chift committed:
chore: update dependencies and refactor MCP entry point

- Refactored the main entry point in `__main__.py` to improve structure and clarity.
- Renamed `main` function to `run_mcp_async` for better readability and added a new `main` function as the CLI entry point.

pldesch-chift committed:
refactor: update MCP entry point and remove obsolete main module

- Changed the entry point for the `chift-mcp-server` script to point to `chift_mcp.run_local:main`.
- Removed the `__main__.py` file as it is no longer needed, streamlining the codebase.

pldesch-chift committed:
feat: enhance configuration management and middleware functionality

- Bring back DEFAULT_CONFIG in constants for streamlined domain operation configuration.
- Update Chift class to include function_config, allowing dynamic operation management.
- Refactored UserAuthMiddleware to utilize function_config for consumer-specific operations.
- Enhanced validate_config function to ensure valid and deduplicated operation configurations.
- Removed obsolete code and improved logging for better traceability in middleware and utility functions.

pldesch-chift committed:
refactor: update MCP entry point

- Changed the entry point for the `chift-mcp-server` script to point to `chift_mcp.main:main`.

pldesch-chift committed:
chore: rename project and update version

- Renamed the project from `chift-mcp-server` to `chift-mcp`.
- Updated version from `0.1.4` to `0.2.0` in `pyproject.toml` and `uv.lock`.
- Introduced a new `__init__.py` file to define the public API for the package.
- Refactored middleware to use `EnvAuthMiddleware` instead of `UserAuthMiddleware`.
- Enhanced MCP initialization by creating a new `create_mcp` function for better configuration management.

pldesch-chift committed:
feat: enhance project configuration and documentation

- Added `fastmcp` as a source in `pyproject.toml` for improved functionality.
- Updated `README.md` to include new sections for E-commerce and Banking, detailing available operations.
- Refined configuration examples in the documentation for clarity.
- Improved test setup by modifying middleware initialization in tests for better coverage of edge cases.

pldesch-chift committed:
chore: update fastmcp source to use Git repository instead of local file

- Changed the `fastmcp` source in `pyproject.toml` and `uv.lock` to point to the Git repository instead of a local directory.
- Updated the versioning and source specification for `fastmcp` to ensure proper dependency management.

pldesch-chift committed:
refactor: update fastmcp configuration and entry point

- Changed the source path in `fastmcp.json` to point to `src/chift_mcp/__main__.py` and updated the entrypoint to `configure_mcp`.
- Updated the script entry in `pyproject.toml` to reflect the new main module.
- Removed the obsolete `main.py` file to streamline the codebase and improve clarity.

pldesch-chift committed:
refactor: clean up remove unused tool filtering

pldesch-chift committed:
feat: add server_name configuration to Chift class

- Introduced a new `server_name` attribute to the `Chift` class in `config.py` for better identification of the MCP server.
- Removed unnecessary logging statements in `FilterToolsMiddleware` to clean up the code and improve performance.

pldesch-chift committed:
feat: integrate proxy functionality into MCP configuration to add SearchChift tool

pldesch-chift committed:
fix: correct comment formatting in FilterToolsMiddleware for consistency

pldesch-chift committed:
refactor: update README and implement direct experimental FastMCPOpenAPI usage to bypass useless env variable

pldesch-chift committed:
fix: update transport method in MCP to use 'stdio' and remove redundant entry in README

pldesch-chift committed:
refactor: update MCP configuration to use Chift class for authentication

- Modified the `create_mcp` function to accept `url_base` and `chift_config` parameters.
- Refactored `ClientAuth` to utilize the `Chift` configuration for authentication details.
- Ensured proper handling of `url_base` and `chift_config` in the HTTP client setup.

pldesch-chift committed:
feat: add timeout parameter to create_mcp function for improved request handling

pldesch-chift committed:
refactor(test): change testing on tool listing

dimaredchyts reviewed (changes requested):
Thanks for your work. I've added a few small comments.

hhertoghe commented on src/chift_mcp/http_client.py:
```diff
@@ -0,0 +1,30 @@
+    chift_config: Chift | None,
+    url_base: str,
+) -> AsyncClient:
+    return AsyncClient(base_url=url_base, auth=ClientAuth(chift_config) if chift_config else None)
```
When run from the mcp-server, we do not need this.

pldesch-chift commented on pyproject.toml:
```diff
@@ -68,6 +70,16 @@ skip-magic-trailing-comma = false
+python_functions = ["test_*"]
+
+[tool.uv.sources]
+fastmcp = { git = "https://github.com/jlowin/fastmcp.git" }
```
Not anymore, they just released version 2.12 that fixes the issue we had

pldesch-chift commented on src/chift_mcp/http_client.py:
```diff
@@ -0,0 +1,30 @@
+
+
+def get_http_client(
+    chift_config: Chift | None,
```
In remote mcp only

dimaredchyts replied:
Yes, got it now

pldesch-chift commented on README.md:
```diff
@@ -199,7 +200,8 @@ env = {
     "CHIFT_ACCOUNT_ID": "your_account_id",
-    "CHIFT_URL_BASE": "https://api.chift.eu"
+    "CHIFT_URL_BASE": "https://api.chift.eu",
+    "CHIFT_CONSUMER_ID": "your_consumer_id",
```
Later on in the readme you have already :

When `CHIFT_CONSUMER_ID` is set, the server lists only the tools relevant to that consumer's connection types. If it's not set (local stdio usage), a small discovery set of tools is exposed to fetch the available consumers and related connectors.

Usefull to add more ?

pldesch-chift commented on fastmcp.json:
```diff
@@ -0,0 +1,15 @@
+{
```
You can launch the MCP server by just running `cd /path/to/dir && fastmcp run`

pldesch-chift committed:
refactor: update MCP configuration and improve client authentication

- Removed the deprecated `mcp.py` file and refactored the `create_mcp` function to utilize the new `get_proxy` method for proxy handling.
- Updated the `Chift` class to use `SecretStr` for sensitive fields and added a default `proxy_url`.
- Enhanced the `ClientAuth` class to accept individual parameters for authentication instead of a `Chift` instance.
- Adjusted the `get_http_client` function to enforce configuration checks based on remote/local mode.
- Improved the `FilterToolsMiddleware` to handle tool name validation more robustly.

dimaredchyts commented on README.md:
```diff
@@ -378,3 +378,140 @@ You can customize this configuration by setting the `CHIFT_FUNCTION_CONFIG` envi
+uv run pytest
+
+# Run linting
+uv run poe lint
```
i think you can run just like `poe lint`, no? same with pytest

pldesch-chift replied:
Done

pldesch-chift committed:
docs: update README to simplify pytest command usage

dimaredchyts reviewed (approved):
LGTM, thanks!
